src/System.Linq/src/System/Linq/Set.cs

```diff
@@ -191,6 +191,38 @@ internal List<TElement> ToList()
         }
 
         /// <summary>
+        /// Converts this set to a <see cref="HashSet{T}"/>.
+        /// </summary>
+        /// <param name="comparer">The equality comparer for the new <see cref="HashSet{T}"/>.</param>
+        /// <returns>A <see cref="HashSet{T}"/> of the items in this set.</returns>
+        internal HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer)
```

Comment:
You can remove this method and change all of the callsites to use `new HashSet<T>(this, comparer)`. Again, if someone wants to do e.g. a union operation they can use HashSet's UnionWith, there's no good reason for someone to do `Union.ToHashSet` and the added complexity does not seem to be worth it.